loadgen/loop.js

```diff
@@ -3,24 +3,24 @@
 
 import { performance } from 'perf_hooks';
 import http from 'http';
-import { prepareFaucet } from './task-tap-faucet';
+// import { prepareFaucet } from './task-tap-faucet';
 import { prepareAMMTrade } from './task-trade-amm';
 import { prepareVaultCycle } from './task-create-vault';
 
 // we want mostly AMM tasks, and only occasional vault tasks
 
 let currentConfig = {
-  faucet: null, // or { interval=60, limit=1, wait=0 }
+  // faucet: null, // or { interval=60, limit=1, wait=0 }
   amm: null,
   vault: null,
   // amm: { interval: 120},
   // vault: { interval: 120, wait: 60 },
 };
 
 const tasks = {
-  faucet: [prepareFaucet],
-  // we must start the AMM task before Vault, because it sells BLD, and both
-  // measure the balance
+  // faucet: [prepareFaucet],
```

Comment:
@katelynsills or @michaelfig Is there an easy way to query a known location on the chain, to grab a copy of the pre-deployed faucet contract? I think the `board` is relevant here? We can land this PR as-is, but it'd be nice to enable faucet-tap loadgen again, and to do that when we've got 100+ load generators means we need to prevent them from all installing their own copies of the contract.

I'm thinking that `task-tap-faucet` should check the `board` at a pre-configured name, and if present, use that as the contract public facet. Then we should write a separate deploy script which installs that contract and publishes the pointer to the `board`.

Reply:
Yup, this could be divided into a step to install the faucet contract (only run once per chain) and a step to start a contract instance. This is the normal split we have for contract/deploy.js vs api/deploy.js in the dapps. Or, did you mean that everyone should be using the same instance as well? 

Either way, both the instance or installation could be put into the board, and a string identifier will be returned. This string can be shared to others. Not sure what mechanism (something through firebase?) exists for sharing that kind of data automatically. But yeah, the loadgen code itself can get the instance or installation given the string identifier by going through the board. To get the public facet, there's an additional call needed to Zoe that passes the instance in. Note that calling `E(zoe).getPublicFacet` will charge a small fee to the feePurse. 

We could put the publicFacet itself in the board, but that's not forwards-compatible with the metering/fee design in which every user gets their own publicFacet object that is bound to the fee purse they used to get it. 

Here's an example of the installation being stored in the board: https://github.com/Agoric/dapp-fungible-faucet/blob/e86e023c553d047188db0b411c689c97bfabcf85/contract/deploy.js#L92

And an example of the installation being retrieved from the board: https://github.com/Agoric/dapp-fungible-faucet/blob/e86e023c553d047188db0b411c689c97bfabcf85/api/deploy.js#L80

Reply:
Could the client hash the source and use that as a key? Actually feels like something that could be done for them automatically for bundle source, a little less so for single installations.

Reply:
Blah I didn't think. The key is auto generated and there is obviously no public write place where key is user controlled....